docs_src/examples/Ch-OSImageWithEdgeX.md

```diff
@@ -766,72 +615,46 @@ Refer to the following to:
-    ```
+List of snaps:
+``` title="🚀 Ubuntu Core"
+$ snap list
+Name                           Version          Rev    Tracking       Publisher   Notes
+core20                         20220805         1611   latest/stable  canonical✓  base
+core22                         20220607         188    latest/stable  canonical✓  base
+edgex-config-provider-example  2.3              2      latest/edge    farshidtz   -
+edgex-device-virtual           2.3.0            335    latest/edge    canonical✓  -
+edgexfoundry                   2.3.0            4101   latest/edge    canonical✓  -
+pc                             20-0.4           x1     -              -           gadget
+pc-kernel                      5.4.0-124.140.1  1077   20/stable      canonical✓  kernel
+snapd                          2.56.2           16292  latest/stable  canonical✓  snapd
+```
+Note that we now also have `edgex-config-provider-example` in the list.
+
+Verify that Device Virtual has the startup message overridden via the gadget defaults:
+``` title="🚀 Ubuntu Core"
+$ snap logs -n=all edgex-device-virtual | grep "Startup message"
+2023-05-25T10:24:50Z edgex-device-virtual.device-virtual[2924]: level=INFO ts=2023-05-25T10:24:50.447466922Z app=device-virtual source=variables.go:457 msg="Variables override of 'Service/StartupMsg' by environment variable: SERVICE_STARTUPMSG=Startup message from gadget!"
+2023-05-25T10:25:03Z edgex-device-virtual.device-virtual[3136]: level=INFO ts=2023-05-25T10:25:03.761993667Z app=device-virtual source=message.go:55 msg="Startup message from gadget!"
+```
 
-Query the metadata of Device Virtual from your host machine. 
-We have to use the same JWT created in chapter B.
+From the host machine, query the device metadata to ensure that Device Virtual has registered only a single virtual device: 
 ```bash title="🖥 Desktop"
-curl --insecure --silent --show-err https://localhost:8443/core-data/api/v2/reading/all?limit=2 -H "Authorization: Bearer $(cat admin-jwt.txt)" | jq
-```
-```json title="Response"
-{
-  "apiVersion": "v2",
-  "statusCode": 200,
-  "totalCount": 133,
-  "readings": [
-    {
-      "id": "f6a53b5c-045f-4913-ae45-4e32642f6102",
-      "origin": 1660923144514370300,
-      "deviceName": "Random-Float-Device",
-      "resourceName": "Float64",
-      "profileName": "Random-Float-Device",
-      "valueType": "Float64",
-      "value": "1.436784e+308"
-    },
-    {
-      "id": "95b5aa9c-e80d-488c-ab5d-1b625a9d0f76",
-      "origin": 1660923114513963300,
-      "deviceName": "Random-Float-Device",
-      "resourceName": "Float64",
-      "profileName": "Random-Float-Device",
-      "valueType": "Float64",
-      "value": "7.737701e+307"
-    }
-  ]
-}
+$ curl --no-progress-meter  http://localhost:59881/api/v3/device/all | jq '.devices[].name'
+"Random-Float-Device"
```

Comment:
Using the edgex-config-provider-example snap from `edge` channel, the output is:
```
"Random-Float-Device"
"Random-Boolean-Device"
"Random-Binary-Device"
"Random-Integer-Device"
"Random-UnsignedInteger-Device"
```
Infos:
```
$ snap list
Name                           Version                   Rev    Tracking          Publisher   Notes
core20                         20230503                  1891   latest/stable     canonical✓  base
core22                         20230503                  634    latest/stable     canonical✓  base
curl                           8.0.1                     1679   latest/stable     woutervb    -
edgex-config-provider-example  v3.0.0-beta+git6.1778bd4  29     latest/edge       farshidtz   -
edgex-device-virtual           3.0.0-dev.51              673    latest/edge       canonical✓  -
edgexfoundry                   3.0.0-dev.164             4455   latest/edge       canonical✓  -
pc                             22-0.3                    x1     -                 -           gadget
pc-kernel                      5.15.0-71.78.1            1281   22/stable         canonical✓  kernel
snapd                          2.59.4                    19361  latest/candidate  canonical✓  snapd
```

Reply:
If this is the case, the config provider has not been connected.

Reply:
It is possible that the connections in gadget.yaml were missing when I did the testing.